onboard/nav/stateMachine.cpp

```diff
@@ -489,7 +489,7 @@ string StateMachine::stringifyNavState() const
 // Returns true if an obstacle is detected, false otherwise.
 bool StateMachine::isObstacleDetected() const
 {
-    return mPhoebe->roverStatus().obstacle().detected;
+    return mPhoebe->roverStatus().obstacle().distance >= 0;
 } // isObstacleDetected()
```

Comment:
Duplicated code, see note above.

Reply:
Technically the inputs are different but functionally they are not. Passing in `this` should not be necessary. If you want to discuss this further, we can collaborate offline.